### PR TITLE
Introduce some pure components.

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7915/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7922/

**Comment by Lukewh** [comment]
Quick QA looks good

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7941/

File: jujugui/static/gui/src/app/components/spinner/spinner.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 'use strict';
 
-class Spinner extends React.Component {
+class Spinner extends React.PureComponent {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/spinner/spinner.js]: I propose that this should be a function not a component

Comment (frankban) [inline jujugui/static/gui/src/app/components/spinner/spinner.js]: Possibly, but why in this branch? Also, that would require an update on the isnpect-component script.

**Comment by frankban** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**